### PR TITLE
#154440552 Implement Products Load Upon Scroll

### PR DESCRIPTION
**Pull request by daddychukz**
#### What does this PR do?
Displays more products from database upon scroll by the user

#### Description of Task to be completed?
 * Write functions to display a limited number of products upon user scroll

#### How should this be manually tested?
Have the app working by:
 * run the command `reaction` on the command line to start the server
 * once app is running, navigate to the app using `localhost:3000`
* Scroll through the product list and you shall observe that products gets loaded automatically upon scroll

 #### Any background context you want to provide?
 Nil

 #### What are the relevant pivotal tracker stories?
#154440552

 #### Screenshots (if appropriate)
screen shot 2018-01-23 at 1 48 36 pm

### CODE REVIEW

File: imports/plugins/core/ui-navbar/client/components/brand.js [hunk 1]
```diff
@@ -23,10 +23,11 @@ class Brand extends Component {
       <a className="brand" onClick={this.handleClick}>
         {this.props.logo &&
           <div className="logo">
-            <img src={this.props.logo} />
+            {<img src={this.props.logo} />}
```
Comment (sachungo) [inline imports/plugins/core/ui-navbar/client/components/brand.js]: Why the outer braces? The previous version is correct. I.e.
```js
<img src={this.props.logo} />

File: imports/plugins/core/ui-navbar/client/components/brand.js [hunk 2]
```diff
@@ -23,10 +23,11 @@ class Brand extends Component {
...
           </div>
         }
-        <span className="title">{this.props.title}</span>
+        <img src="/resources/rc-logo.png" width="70" />
+        {/* <span className="title">{this.props.title}</span> */}
```
Comment (sachungo) [inline imports/plugins/core/ui-navbar/client/components/brand.js]: Delete if it's not used anymore

File: imports/plugins/included/product-variant/components/productGrid.js [hunk 3]
```diff
@@ -1,11 +1,43 @@
...
 import { Components } from "@reactioncommerce/reaction-components";
 
 class ProductGrid extends Component {
   static propTypes = {
     products: PropTypes.array
   }
+  constructor(props) {
+    super(props);
+    this.state = {
+      allProducts: this.props.products,
```
Comment (sachungo) [inline imports/plugins/included/product-variant/components/productGrid.js]: Use `props.products`

File: imports/plugins/included/product-variant/components/productGrid.js [hunk 4]
```diff
@@ -1,11 +1,43 @@
...
+
+  componentWillReceiveProps(nextProps) {
+    if (this.state.allProducts.length > 12) {
+      this.setState({ nextProducts: nextProps.products.slice(0, this.state.lastIndex) });
+    } else {
+      this.setState({ nextProducts: this.state.allProducts });
+    }
+  }
+
+  generateProducts() {
```
Comment (sachungo) [inline imports/plugins/included/product-variant/components/productGrid.js]: Write this as 
```js
generateProducts = () => {...}
```
then you won't have to bind the method in the constructor

File: imports/plugins/included/product-variant/components/productGrid.js [hunk 5]
```diff
@@ -29,14 +61,22 @@ class ProductGrid extends Component {
     );
   }
 
+
```
Comment (sachungo) [inline imports/plugins/included/product-variant/components/productGrid.js]: Delete extra line

File: imports/plugins/included/product-variant/components/products.js [hunk 6]
```diff
@@ -66,6 +66,7 @@ class Products extends Component {
         productIds={getIds({ tags: products })}
         canEdit={Reaction.hasPermission("createProduct")}
         products={products}
+        /* loadMoreProducts={this.props.loadProducts(event)} */
```
Comment (sachungo) [inline imports/plugins/included/product-variant/components/products.js]: Is it still needed in future code? If not, delete the line

File: imports/plugins/included/product-variant/components/products.js [hunk 7]
```diff
@@ -133,7 +134,7 @@ class Products extends Component {
         return (
           <div id="container-main">
             {this.renderProductGrid()}
-            {this.renderLoadMoreProductsButton()}
+            {/* this.renderLoadMoreProductsButton() */}
```
Comment (sachungo) [inline imports/plugins/included/product-variant/components/products.js]: Delete it

File: imports/plugins/included/product-variant/containers/productGridContainer.js [hunk 8]
```diff
@@ -151,6 +151,7 @@ const wrapComponent = (Comp) => (
             canEdit={this.props.canEdit}
             isSearch={this.props.isSearch}
             unmountMe={this.props.unmountMe}
+            /* loadMoreProducts={this.props.loadMoreProducts} */
```
Comment (sachungo) [inline imports/plugins/included/product-variant/containers/productGridContainer.js]: Delete

**Review by sachungo — CHANGES_REQUESTED** [review]
Minor feedback to be actioned

File: imports/plugins/core/ui-navbar/client/components/brand.js (hunk 1 above)
Comment (daddychukz) [reply imports/plugins/core/ui-navbar/client/components/brand.js]: Fixed

File: imports/plugins/core/ui-navbar/client/components/brand.js (hunk 2 above)
Comment (daddychukz) [reply imports/plugins/core/ui-navbar/client/components/brand.js]: Deleted

File: imports/plugins/included/product-variant/components/productGrid.js (hunk 3 above)
Comment (daddychukz) [reply imports/plugins/included/product-variant/components/productGrid.js]: Changed

File: imports/plugins/included/product-variant/components/productGrid.js (hunk 4 above)
Comment (daddychukz) [reply imports/plugins/included/product-variant/components/productGrid.js]: Refactored

File: imports/plugins/included/product-variant/components/productGrid.js (hunk 5 above)
Comment (daddychukz) [reply imports/plugins/included/product-variant/components/productGrid.js]: Deleted

File: imports/plugins/included/product-variant/components/products.js (hunk 6 above)
Comment (daddychukz) [reply imports/plugins/included/product-variant/components/products.js]: Deleted

File: imports/plugins/included/product-variant/components/products.js (hunk 7 above)
Comment (daddychukz) [reply imports/plugins/included/product-variant/components/products.js]: Deleted

File: imports/plugins/included/product-variant/containers/productGridContainer.js (hunk 8 above)
Comment (daddychukz) [reply imports/plugins/included/product-variant/containers/productGridContainer.js]: Deleted

File: imports/plugins/included/product-variant/components/productGrid.js [hunk 9]
```diff
@@ -1,11 +1,42 @@
...
   }
+  constructor(props) {
+    super(props);
+    this.state = {
+      allProducts: props.products,
+      nextProducts: [],
+      lastIndex: 12,
+      hasMore: true
+    };
+    // this.generateProducts = this.generateProducts.bind(this);
```
Comment (sachungo) [inline imports/plugins/included/product-variant/components/productGrid.js]: No need for this line. Please delete

Comment (daddychukz) [reply imports/plugins/included/product-variant/components/productGrid.js]: deleted

**Review by sachungo — APPROVED** [review]

**Pull request merged**